package/MDAnalysis/analysis/rms.py

```diff
@@ -585,40 +567,27 @@ def run(self, start=None, stop=None, step=None, progout=10, quiet=False):
         .. [Welford1962] B. P. Welford (1962). "Note on a Method for
            Calculating Corrected Sums of Squares and Products." Technometrics
            4(3):419-420.
-        """
-        traj = self.atomgroup.universe.trajectory
-        start, stop, step = traj.check_slice_indices(start, stop, step)
-        sumsquares = np.zeros((self.atomgroup.n_atoms, 3))
-        means = np.array(sumsquares)
-
-        if quiet:
-            progout = None
-
-        # set up progress output
-        if progout:
-            percentage = ProgressMeter(self.atomgroup.universe.trajectory.n_frames,
-                                       interval=progout)
-        else:
-            percentage = ProgressMeter(self.atomgroup.universe.trajectory.n_frames,
-                                       quiet=True)
+    """
+    def __init__(self, atomgroup, weights=None, **kwargs):
+        super(RMSF, self).__init__(atomgroup.universe.trajectory, **kwargs)
+        self.atomgroup = atomgroup
+        if weights == 'mass':
+            weights = self.atomgroup.masses
+        self.weights = weights
 
-        for k, ts in enumerate(self.atomgroup.universe.trajectory[start:stop:step]):
-            sumsquares += (k/(k + 1.0)) * (self.atomgroup.positions - means)**2
-            means = (k * means + self.atomgroup.positions)/(k + 1)
+    def _prepare(self):
+        self.sumsquares = np.zeros((self.atomgroup.n_atoms, 3))
+        self.mean = self.sumsquares.copy()
 
-            percentage.echo(ts.frame)
+    def _single_frame(self):
+        k = self._frame_index
+        self.sumsquares += (k / (k+1.0)) * (self.atomgroup.positions - self.mean) ** 2
+        self.mean = (k * self.mean + self.atomgroup.positions) / (k + 1)
 
-        rmsf = np.sqrt(sumsquares.sum(axis=1)/(k + 1))
+    def _conclude(self):
+        k = self._frame_index
+        self.rmsf = np.sqrt(self.sumsquares.sum(axis=1) / (k + 1))
 
-        if not (rmsf >= 0).all():
+        if not (self.rmsf >= 0).all():
             raise ValueError("Some RMSF values negative; overflow " +
                              "or underflow occurred")
-
-        self._rmsf = rmsf
-
-    @property
-    def rmsf(self):
```

Comment:
This isn't removed. It can be read the same way as before. Just that you can now also write to the value. So there isn't a change for the user. We handle all analysis classes right now that computed values can also be overwritten. So this is just making it behave more like the rest of MDAnalysis.